Sources/StreamChat/ChatClient.swift

```diff
@@ -379,8 +356,16 @@ public class ChatClient {
     func createBackgroundWorkers() {
         backgroundWorkers = workerBuilders.map { builder in
             builder(self.databaseContainer, self.apiClient)
-        } + eventWorkerBuilders.map { builder in
-            builder(self.databaseContainer, self.eventNotificationCenter, self.apiClient)
+        }
+        
+        if let webSocketClient = webSocketClient {
```

Comment:
`if let` usually decreases the clarity/adds complexity when reading the code. 

Could we use `guard let` instead?

Reply:
Yep, thanks for the question! There was the reason to use `if`, actually 🙂 I prefer to use `guard` when further execution should not happen because of some condition not being met. In this case absence of `webSocketClient` does not mean we should return from `createBackgroundWorkers` because there can be other workers that do not rely on the web-socket but initialized after `connectionRecoveryHandler`